xwayland/xwm.c

```diff
@@ -187,44 +300,6 @@ static void read_surface_parent(struct wlr_xwm *xwm,
 	wl_signal_emit(&surface->events.set_parent, surface);
 }
 
-static void handle_surface_state(struct wlr_xwm *xwm,
```

Comment:
Why is this broken?

Reply:
Now that I'm looking at it again, it looks ok. Something was setting a bunch of junk in the window property that was making windows think they are maximized, so I just blamed this code and removed it.

There are several things wrong with this approach in general though.

1. It emits events for state changes we don't care about. Many of these don't translate to wayland well and require a level of knowledge of X11 to interpret I wouldn't expect library users to have. It's best to keep this interface slim or else it will become hard to maintain. This is fine because clients are supposed to read `_NET_SUPPORTED` from the root window and not send us events that aren't in there, but some toolkits (GTK) ignore this property and send us the events anyway.
2. It's inconsistent with the other shell apis and will be a bit cumbersome to abstract into a view properly.